libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs

```diff
@@ -299,13 +290,9 @@ public async Task<TokenResponse> GetUserTokenAsync(DialogContext dc, Cancellatio
         /// <returns>A task that represents the work queued to execute.</returns>
         public async Task SignOutUserAsync(DialogContext dc, CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (!(dc.Context.Adapter is IUserTokenProvider adapter))
-            {
-                throw new InvalidOperationException("OAuthPrompt.SignOutUser(): not supported by the current adapter");
-            }
-
             // Sign out user
-            await adapter.SignOutUserAsync(dc.Context, ConnectionName.GetValue(dc.State), dc.Context.Activity?.From?.Id, cancellationToken).ConfigureAwait(false);
+            var settings = new OAuthPromptSettings { ConnectionName = ConnectionName?.GetValue(dc.State) };
```

Comment:
Same here, what's the valid scenario where ConnnectionName is null?

Reply:
The null is handled in the next layer(s) - with an ArgumentNullException(nameof(connectionName)) which seems like the right exception.